resources/assets/coffee/react/_mixins/scrolling-page.coffee

```diff
@@ -0,0 +1,82 @@
+###
+# Copyright 2016 ppy Pty. Ltd.
+#
+# This file is part of osu!web. osu!web is distributed with the hope of
+# attracting more community contributions to the core ecosystem of osu!.
+#
+# osu!web is free software: you can redistribute it and/or modify
+# it under the terms of the Affero GNU General Public License version 3
+# as published by the Free Software Foundation.
+#
+# osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+
+pages = document.getElementsByClassName("js-switchable-mode-page--scrollspy")
+pagesOffset = document.getElementsByClassName("js-switchable-mode-page--scrollspy-offset")
+
+@ScrollingPageMixin =
+  componentWillUnmount: ->
+    clearTimeout @modeScrollTimeout
+
+  setCurrentMode: (_e, mode) ->
```

Comment:
this should be removed from this mixin. Current `master` branch added additional stuff for this function in `ProfilePage.Main` as well, making abstracting it much less useful.

If anything, something like `SetOneStateMixin` mixin is more useful.

```coffeescript
# rough idea, has not been actually tested
@SetOneStateMixin =
  setOneState: (key, newValue, callback) ->
    return callback?() if @state[key] == newValue

    @setState "#{key}": newValue, callback
```